solr/solrj/src/test/org/apache/solr/client/solrj/impl/Http2SolrClientTest.java

```diff
@@ -69,7 +69,7 @@ public void testTimeout() throws Exception {
       client.query(q, SolrRequest.METHOD.GET);
       fail("No exception thrown.");
     } catch (SolrServerException e) {
-      assertTrue(e.getMessage().contains("timeout") || e.getMessage().contains("Timeout"));
+      assertTrue(isTimeout(e));
```

Comment:
For asserting on `Exceptions` could we `assertThrows` throws instead? Then assert that it was a timeout after? I like that over asserting on a catch on all these tests and the ones you updated below. Kinda of like this

```suggestion
SolrServerException exception = assertThrows("Expected an exception but no expcetion thrown", SolrServerException.class, () -> {
      // Run your logic to throw an excpetion
      });
 assertTrue("Expected a timeout exception but got: " + exception.getMessage(), isTimeout(exception));
```